java-checks-test-sources/src/main/java/checks/StaticMembersAccessCheck.java

```diff
@@ -33,18 +36,38 @@ public StaticMembersAccessCheckD d() {
   }
 
   public void noncompliant() {
-    first.counter ++; // Noncompliant [[sc=5;ec=18]] {{Change this instance-reference to a static reference.}}
+    first.counter ++; // Noncompliant [[sc=5;ec=18;quickfixes=qf1]] {{Change this instance-reference to a static reference.}}
+    // fix@qf1 {{Replace "first" by "StaticMembersAccessCheckA"}}
+    // edit@qf1 [[sc=5;ec=10]] {{StaticMembersAccessCheckA}}
     second.counter ++; // Noncompliant
-    second.method(); // Noncompliant
+    second.method(); // Noncompliant [[sc=5;ec=18;quickfixes=qf2]]
+    // fix@qf2 {{Replace "second" by "StaticMembersAccessCheckA"}}
+    // edit@qf2 [[sc=5;ec=11]] {{StaticMembersAccessCheckA}}
     third.counter ++; // Noncompliant
     first.d.counter++; // Noncompliant
-    first.c.d.counter++; // Noncompliant
-    first.d().counter++; // Noncompliant
+    first.c.d.counter++; // Noncompliant [[sc=5;ec=22;quickfixes=qf3]]
+    // fix@qf3 {{Replace "first.c.d" by "StaticMembersAccessCheckD"}}
+    // edit@qf3 [[sc=5;ec=14]] {{StaticMembersAccessCheckD}}
+    first.d().counter++; // Noncompliant // Noncompliant [[sc=5;ec=22;quickfixes=qf4]]
+    // fix@qf4 {{Replace "first.d()" by "StaticMembersAccessCheckD"}}
+    // edit@qf4 [[sc=5;ec=14]] {{StaticMembersAccessCheckD}}
     d().counter++; // Noncompliant
-    ((StaticMembersAccessCheckA.StaticMembersAccessCheckD) d()).counter++; // Noncompliant [[sc=5;ec=72]]
+    // Noncompliant@+1 [[sc=5;el=+3;ec=14;quickfixes=qf5]]
+    (
+      (StaticMembersAccessCheckA.StaticMembersAccessCheckD) d()
+    ).counter++;
+    // fix@qf5 {{Replace "( (StaticMembersAccessCheckA.StaticMembersAccessCheckD) d() )" by "StaticMembersAccessCheckD"}}
```

Comment:
quickfix messages should not be too long. I feel here it can escalate fastly. What about using the following message instead when it's not a "simple case": `Replace the expression by "StaticMembersAccessCheckD"` I feel it would make it way nicer.